### PR TITLE
[vividus-mobile-app] Add step to upload file with data to the device

### PR DESCRIPTION
**Pull request by IBrun**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-mobile-app.adoc [hunk 1]
```diff
@@ -441,6 +441,28 @@ When I upload file `images/avatar.png` to device
...
+
+[source,gherkin]
+----
+I upload file with name `$fileName` and data `$data` to device
+----
+
+* `$data` - The data to store as a file.
+* `$fileName` - The name of the uploaded file.
+
+.Upload file with image data to the device
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mobile-app.adoc:457]: please add one more example showing how to save text data

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/DeviceActions.java [hunk 2]
```diff
@@ -48,8 +48,17 @@ public DeviceActions(IWebDriverProvider webDriverProvider, WebDriverSessionInfo
...
-        byte[] base64Bytes = Base64.getEncoder().encode(fileBytes);
+        pushBinaryData(deviceFilePath, fileBytes);
+    }
 
+    /**
+     * Pushes <b>binary data</b> onto a device at the <b>deviceFilePath</b> path
+     * @param deviceFilePath The destination file path on a device.
+     * @param data The binary data to push.
+     */
+    public void pushBinaryData(String deviceFilePath, byte[] data)
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/DeviceActions.java:59]:
```suggestion
    public void pushFile(String deviceFilePath, byte[] data)
```
also javadoc should be reworked

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/steps/DeviceSteps.java [hunk 3]
```diff
@@ -53,6 +54,21 @@ public void uploadFileToDevice(String filePath)
...
 
+    /**
+     * Upload the file with data to the device
+     * @param data The data to store as a file.
+     * @param fileName The name of the uploaded file.
+     */
+    @When("I upload file with name `$fileName` and data `$data` to device")
+    public void uploadFileWithDataToDevice(String fileName, DataWrapper data)
+    {
+        LOGGER.info("Uploading file with name `{}` and data to the device at '{}' folder", fileName,
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/steps/DeviceSteps.java:65]:
```suggestion
        LOGGER.info("Uploading file with name `{}` to the device at '{}' folder", fileName,
```

File: docs/modules/plugins/pages/plugin-mobile-app.adoc [hunk 4]
```diff
@@ -441,6 +441,28 @@ When I upload file `images/avatar.png` to device
...
+
+Upload the file with data to the device/emulator/simulator.
+
+[source,gherkin]
+----
+I upload file with name `$fileName` and data `$data` to device
+----
+
+* `$data` - The data to store as a file.
+* `$fileName` - The name of the uploaded file.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mobile-app.adoc:455]: add a reference on how to find/configure folder for upload

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/steps/DeviceSteps.java [hunk 5]
```diff
@@ -53,6 +54,21 @@ public void uploadFileToDevice(String filePath)
...
+                folderForFileUpload);
+
+        String deviceFilePath = Paths.get(folderForFileUpload, fileName).toString();
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/steps/DeviceSteps.java:68]: please refactor to avoid code duplications

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-mobile-app.adoc [hunk 6]
```diff
@@ -441,6 +441,28 @@ When I upload file `images/avatar.png` to device
...
 ----
 
+
+=== Upload the file with data to the device
+
+Upload the file with data to the device/emulator/simulator.
+
+[source,gherkin]
+----
+I upload file with name `$fileName` and data `$data` to device
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-mobile-app.adoc:451]: Keyword is missing

File: docs/modules/plugins/pages/plugin-mobile-app.adoc [hunk 7]
```diff
@@ -441,6 +441,28 @@ When I upload file `images/avatar.png` to device
...
+
+* `$data` - The data to store as a file.
+* `$fileName` - The name of the uploaded file.
+
+.Upload file with image data to the device
+[source,gherkin]
+----
+When I execute HTTP GET request for resource with URL `https://raw.githubusercontent.com/vividus-framework/vividus/master/vividus-tests/src/main/resources/baselines/context.png`
+When I upload file with name `context.png` and data `${response-as-bytes}` to device
+Then the image is displayed with the uploaded image
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-mobile-app.adoc:462]:
```suggestion
Then image is displayed in gallery
```

File: docs/modules/plugins/partials/mobile-upload-folder-info.adoc [hunk 8]
```diff
@@ -0,0 +1,2 @@
+[NOTE]
+The destination folder on the device for the uploaded file is configured by `mobile-environment.devic.folder-for-file-upload` property. See more information in the <<mobile_properties,properties section.>>
```
Comment (valfirst) [inline docs/modules/plugins/partials/mobile-upload-folder-info.adoc:2]:
```suggestion
The destination folder on the device for the uploaded file is configured by `mobile-environment.device.folder-for-file-upload` property. See more information in the <<mobile_properties,properties section.>>
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**